expediagroup-sdk-rest/src/test/kotlin/com/expediagroup/sdk/rest/util/UrlUtilsTest.kt

```diff
@@ -0,0 +1,380 @@
+                parseUrlPathTemplateParams(
+                    template = template,
+                    path = path
+                )
+
+            // Then
+            val expected =
+                mapOf(
+                    "userId" to "123",
+                    "postId" to "456"
+                )
+            assertEquals(expected, result)
+        }
+
+        @Test
+        fun `throws illegalArgumentException when template and path do not match`() {
+            // Given
+            val template = "/api/v1/users"
+            val path = "/api/v1/users/123/posts/456"
+
+            // When
+            val exception =
+                assertThrows<IllegalArgumentException> {
+                    parseUrlPathTemplateParams(
+                        template = template,
+                        path = path
+                    )
+                }
+
+            // Then
+            val expectedExceptionMessage =
+                "Template and path segments must have the same number of segments"
+
+            assertEquals(expectedExceptionMessage, exception.message)
+        }
+
+        @Test
+        fun `returns empty map when template and path are identical`() {
+            // Given
+            val template = "/api/v1/users/123/posts/456"
+            val path = "/api/v1/users/123/posts/456"
+
+            // When
+            val result =
+                parseUrlPathTemplateParams(
+                    template = template,
+                    path = path
+                )
+
+            // Then
+            assertEquals(emptyMap<String, String>(), result)
+        }
+
+        @ParameterizedTest
+        @ValueSource(
+            strings = [
+                "{id",
+                "id}",
+                "i{d}",
+                "{i}d"
+            ]
+        )
+        fun `does not parse param when param format is invalid`(paramTemplate: String) {
+            // Given
+            val template = "/api/v1/users/$paramTemplate"
+            val path = "/api/v1/users/123"
+
+            // When
+            val result =
+                parseUrlPathTemplateParams(
+                    template = template,
+                    path = path
+                )
+
+            // Then
+            assertEquals(emptyMap<String, String>(), result)
+        }
+
+        @Test
+        fun `returns empty map when template and path are empty`() {
+            // Given
+            val template = ""
+            val path = ""
+
+            // When
+            val result =
+                parseUrlPathTemplateParams(
+                    template = template,
+                    path = path
+                )
+
+            // Then
+            assertEquals(emptyMap<String, String>(), result)
+        }
+    }
+
+    @Nested
+    inner class ParseUrlQueryParamsFromQueryTest {
+        @Test
+        fun `parses query string with pipe-delimited values using pipe strategy`() {
+            val query = "key=value1|value2|value3"
```

Comment:
Do you see a value in adding one more `key` so we can cover multiple keys scenarios in these tests?
```suggestion
            val query = "key1=value1|value2|value3&key2=value1,|value2"
```

Reply:
Done.